Create email scraper, password cracker, and ssid connector files

astralm0nke opened the pull request:
Created email scraper file in Web directory.

astralm0nke reviewed (commented):
review

dmdhrumilmistry commented on pyhtools/attackers/web/email_scraper.py:
```diff
@@ -0,0 +1,74 @@
+from bs4 import BeautifulSoup
+import requests, re, lxml
+from requests import exceptions
+from urllib import parse
+from collections import deque
+from pyhtools.UI.colors import BRIGHT_RED, BRIGHT_WHITE, BRIGHT_YELLOW
```
python std library imports should be first followed by third party libraries and pyhtools import at the last.

dmdhrumilmistry commented on pyhtools/attackers/web/email_scraper.py:
```diff
@@ -0,0 +1,74 @@
+from collections import deque
+from pyhtools.UI.colors import BRIGHT_RED, BRIGHT_WHITE, BRIGHT_YELLOW
+
+##----------------Author: astralm0nke on GitHub-----------------##
```
Let's remove author comment since you'll appear in contributors list.

dmdhrumilmistry commented on pyhtools/attackers/web/email_scraper.py:
```diff
@@ -0,0 +1,74 @@
+
+##----------------Author: astralm0nke on GitHub-----------------##
+class EmailScraper:
+    USER_URL = str(input('[+] Specify Target URL to Scan: '))
+    USER_COUNT = int(input('[+] Specify Count Limit: '))
+    COUNT = 0
+    URLS = deque([USER_URL])
+    SCRAPPED_URLS = set()
+    EMAILS = set()
```
we should avoid using `input` while writing a class instead we can assign these variables in `__init__ ` method and use `argparse` to accept inputs in `__main__` block.

main objective to follow this is others can use pyhtools as per their requirements.

dmdhrumilmistry commented on pyhtools/attackers/web/email_scraper.py:
```diff
@@ -0,0 +1,74 @@
+    SCRAPPED_URLS = set()
+    EMAILS = set()
+    
+# Main function to scrape emails
+    def scrape(self):
```
Let's remove comment and add docstring in this [format](https://github.com/dmdhrumilmistry/pyhtools/blob/deadf6d0aa04d0e3666cc41e2aaf5527b126f184/pyhtools/attackers/web/utils.py#L39) so others can know how to use this function.

dmdhrumilmistry commented on pyhtools/attackers/web/email_scraper.py:
```diff
@@ -0,0 +1,74 @@
+             print(BRIGHT_RED + '[-] No Emails Found. Womp-Womp!')
+
+# Export emails to a text document, creates file if not found.        
+    def export(self):
```
Let's also add docstring here and remove comment

dmdhrumilmistry commented on pyhtools/attackers/web/email_scraper.py:
```diff
@@ -0,0 +1,74 @@
+
+# Export emails to a text document, creates file if not found.        
+    def export(self):
+        path = str(input('Please specify the path to save file: '))
```
we can accept input as function argument instead of using `input`

dmdhrumilmistry commented on pyhtools/attackers/web/email_scraper.py:
```diff
@@ -0,0 +1,74 @@
+# Export emails to a text document, creates file if not found.        
+    def export(self):
+        path = str(input('Please specify the path to save file: '))
+        from datetime import date
```
we can move this import at the top.

dmdhrumilmistry commented on pyhtools/attackers/web/email_scraper.py:
```diff
@@ -0,0 +1,74 @@
+from bs4 import BeautifulSoup
+import requests, re, lxml
```
do we need to add lxml package in our dependencies list?

astralm0nke replied:
Probably; the scraper won't function without it.

dmdhrumilmistry replied:
^^

astralm0nke reviewed (commented):
review

dmdhrumilmistry commented:
> review

currently there are no new changes pushed

astralm0nke commented:
> > review
> 
> 
> 
> currently there are no new changes pushed  

Can you see the "replace input with argparser in class constructor" and "edit docstrings"  commits on your end? I'll push again when I'm back at my computer today.

astralm0nke reviewed (commented):
I was going to push these in a separate PR, but I saw your message about there being no new changes published; the email_scraper file should contain the edits you requested. Let me know if there are any problems. Thanks!

dmdhrumilmistry commented:
> I was going to push these in a separate PR, but I saw your message about there being no new changes published; the email_scraper file should contain the edits you requested. Let me know if there are any problems. Thanks!

I'm still unable to view changes. Can you remove argparse from classes. We shouldn't use inside the classes since it'll force users to use cli. What you can do is separate argparse from classes and create examples using argparse.

This will allow users to use it wherever they want such as desktop/web applications.

astralm0nke reviewed (commented):
Hello, apologies for the time it took to respond to requests. Let me know if you're still having trouble seeing changes on your end; it should show edits to the 3 files as well as 3 additions to the examples folder. Thanks!

Lefa7 reviewed (commented):
Ok

dmdhrumilmistry commented on pyhtools/attackers/Network/pass_cracker.py:
```diff
@@ -0,0 +1,47 @@
+import hashlib
+from pyhtools.UI.colors import *
```
instead of using `*` you can import specific constants that are required.

dmdhrumilmistry commented on pyhtools/attackers/Network/ssid_connector.py:
```diff
@@ -0,0 +1,42 @@
+import argparse as ap
+from wireless import Wireless
```
^^

astralm0nke replied:
Just wireless. I'll add it to dependencies.

dmdhrumilmistry commented on pyhtools/attackers/Network/ssid_connector.py:
```diff
@@ -0,0 +1,39 @@
+from wireless import Wireless
+from pyhtools.UI.colors import *
```
only import constants that are required.

astralm0nke commented:
Implemented requested changes.

dmdhrumilmistry commented on poetry.lock:
```diff
@@ -1754,6 +1912,16 @@ files = [
 ]
 
+[[package]]
+name = "wireless"
```
this package is not being maintained anymore. Is there any other alternative for this package?

https://pypi.org/project/wireless/

astralm0nke replied:
I think there's a newer package out that does the same thing. I'll edit and push

dmdhrumilmistry replied:
great!!

dmdhrumilmistry reviewed (commented):
it seems like `wireless` package is not being maintained anymore, I would prefer to get it replaced with some other healthy package. Everything else seems fine to me.

astralm0nke commented:
There's a class in that newer package that I'm a little unfamilier with. I removed the ssid_connector in the meantime to work on it locally and updated poetry.

dmdhrumilmistry commented:
Okay, I'll merge changes in this branch for now. You can create a new PR when you're done.
